src/main/kotlin/inspections/variable/VariableNamesInspection.kt

```diff
@@ -0,0 +1,76 @@
+package inspections.variable
+
+import com.intellij.codeInspection.*
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.project.Project
+import com.intellij.psi.JavaElementVisitor
+import com.intellij.psi.PsiElementVisitor
+import com.intellij.psi.PsiVariable
+import com.intellij.psi.SmartPsiElementPointer
+import com.intellij.refactoring.suggested.createSmartPointer
+import inspections.variable.ProbabilitiesStorage.Companion.recalculateLater
+import org.jetbrains.id.names.suggesting.IdNamesSuggestingService
+import org.jetbrains.id.names.suggesting.ModifiedMemberInplaceRenamer
+import org.jetbrains.id.names.suggesting.contributors.NGramVariableNamesContributor.caretInsideVariable
+import org.jetbrains.id.names.suggesting.impl.SuggestVariableNamesIntention
+
+class VariableNamesInspection : AbstractBaseJavaLocalInspectionTool() {
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return VariableVisitor(holder)
+    }
+
+    class VariableVisitor(private val holder: ProblemsHolder) : JavaElementVisitor() {
+        private val probabilityCutoff: Double = 0.001
+
+        override fun visitVariable(variable: PsiVariable?) {
+            when {
+                variable == null -> return
+                ProbabilitiesStorage.ignore(variable) -> return
+                caretInsideVariable(variable) -> recalculateLater(variable)
+                else -> {
+                    if (!ProbabilitiesStorage.contains(variable) || ProbabilitiesStorage.needRecalculate(variable)) {
+                        val prob = IdNamesSuggestingService.getInstance(holder.project).getVariableNameProbability(variable)
+                        ProbabilitiesStorage.put(variable, Probability(prob));
+                    }
+                    val probability = ProbabilitiesStorage.getProbability(variable)
+                    if (probability == null || probability.prob < probabilityCutoff) {
```

Comment:
Looks like a bug. I'd expect something like `probability != null && probability.prob > probabilityCutoff`

Reply:
No, it's not a bug. Identifier name is bad if probability of it is lower than `probabilityCutoff`.